### PR TITLE
(GH-85) Added support for listing commits since a tag

### LINKED ISSUE #85: Log and describe don't work correctly
There is a [simple command in git](https://stackoverflow.com/questions/12082981/get-all-git-commits-since-last-tag) to get every commit since previous tags, something that is useful when creating release notes:

    git log $(git describe --tags --abbrev=0)..HEAD --oneline

The equivalent code for Cake.Git would be

    GitLog(rootDir, GitDescribe(rootDir,  GitDescribeStrategy.Tags));

Unfortunately this does not produce the same result. From the source code it looks like it uses a simple linq expression that stops when it finds the tag, while the [libgit2sharp documentation recommends a different way to do it](https://github.com/libgit2/libgit2sharp/wiki/git-log). For example, given the following git history:

```
        * - HEAD - merge
        |\
        | * - commit1
        | |
        | * - commit2
 Tag2 - * |
        |/
        * - Tag1
        |
```

The command line snippet above lists three lines, the `merge`, `commit1` and `commit2`, but Cake.Git only lists the `merge` commit, since it stops when the linq expression reaches the first tag. 

If Cake.Git instead use `ExcludeReachableFrom` as described in the libgit2sharp documentation it would produce the correct output.

### PR DESCRIPTION
**Pull request by mariusGundersen**
This fixes #85.

### CODE REVIEW

**Comment by mariusGundersen** [comment]
@devlead, any chance you have time to look into this? Or if you are busy, can you delegate the work to someone else? It would be very useful for us to get this fixed

**Comment by megakid** [comment]
I would very much like this feature too!

**Comment by pascalberger** [comment]
@mariusGundersen Thanks for the pull request. To be able to merge it, it needs to be rebased on current `develop` branch. Can you please rebase or allow edits from maintainers on your branch (see [documentation](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)) in which case I would be able to do the rebase myself.

File: src/Cake.Git/GitAliases.Log.cs [hunk 1]
```diff
@@ -142,6 +142,58 @@ string sinceCommitId
                 );
         }
 
+        /// <summary>
+        /// Get commits after a certain tag
+        /// </summary>
+        /// <example>
+        /// <code>
+        ///     var result = GitLog("c:/temp/cake", "since tag");
```
Comment (pascalberger) [inline src/Cake.Git/GitAliases.Log.cs]: Should be `GitLogTag`

File: src/Cake.Git/GitAliases.Log.cs [hunk 2]
```diff
@@ -142,6 +142,58 @@ string sinceCommitId
...
+        /// </code>
+        /// </example>
+        /// <param name="context">The context.</param>
+        /// <param name="repositoryDirectoryPath">Path to repository.</param>
+        /// <param name="sinceTag">Tag to start fetching from.</param>
+        /// <returns>The path to the created repository.</returns>
```
Comment (pascalberger) [inline src/Cake.Git/GitAliases.Log.cs]: Should be `<returns>Commits since the tag passed with <paramref name="sinceTag"/></returns>`

**Review by pascalberger — CHANGES_REQUESTED** [review]

**Comment by mariusGundersen** [comment]
@johnkors can you look into this? I think it would be useful for your build scripts.

**Comment by johnkors** [comment]
@mariusGundersen  Could you allow the maintainers here to edit your branch in your fork, like @pascalberger  said? Seems as he's willing to take it on..?

**Comment by johnkors** [comment]
Ah, sorry. I see, the fork is in _our_ org. Ok, I'll see what I can do!

**Comment by johnkors** [comment]
Looks like I cannot set the checkbox of "let maintainers modify PR" flag, since I was not the author of the PR, so I rebased it.

File: src/Cake.Git/GitAliases.Log.cs (hunk 1 above)
Comment (johnkors) [reply src/Cake.Git/GitAliases.Log.cs]: ✅ Updated in the rebase

File: src/Cake.Git/GitAliases.Log.cs (hunk 2 above)
Comment (johnkors) [reply src/Cake.Git/GitAliases.Log.cs]: ✅ Updated in the rebase

File: test.cake [hunk 3]
```diff
@@ -790,6 +788,25 @@ Task("Git-Clean")
...
+    GitCommit(testInitalRepo, testUser, testUserEmail, "Third commit");
+    CreateRandomDataFile(Context, filePath);
+    GitAdd(testInitalRepo, filePath);
+    GitCommit(testInitalRepo, testUser, testUserEmail, "Fourth commit");
+    CreateRandomDataFile(Context, filePath);
+    GitAdd(testInitalRepo, filePath);
+    GitCommit(testInitalRepo, testUser, testUserEmail, "Fifth commit");
+
+    var commit = GitLogTag(testInitalRepo, "test-tag");
+    Information("{0}", string.Join("\r\n", commit));
```
Comment (pascalberger) [inline test.cake]: Can you change the test so that it actually breaks (throws an exception) if the returned values don't match the expected values.

**Review by pascalberger — CHANGES_REQUESTED** [review]
@johnkors Thanks for the rebase 👍. 

One last thing: The test currently just outputs the result. It would be nice to have an actual test which breaks if another result than the expected is returned.

File: test.cake (hunk 3 above)
Comment (johnkors) [reply test.cake]: ✅ Added validation here https://github.com/cake-contrib/Cake_Git/pull/86/files#diff-fbdef9605bcbb2541e6b786e3666969bR811

**Review by pascalberger — APPROVED** [review]
LGTM

**Comment by pascalberger** [comment]
@mariusGundersen @johnkors  your changes have been merged, thanks for your contribution 👍

**Comment by johnkors** [comment]
No rush, just wondering: any plans for releasing to nuget.org any time soon..?

**Pull request merged**